sdk/src/main/scala/com/horizen/forge/ForgeMessageBuilder.scala

```diff
@@ -3,210 +3,131 @@ package com.horizen.forge
 import akka.util.Timeout
 import com.horizen.block._
 import com.horizen.box.Box
-import com.horizen.chain.{MainchainHeaderHash, SidechainBlockInfo}
 import com.horizen.companion.SidechainTransactionsCompanion
 import com.horizen.consensus._
-import com.horizen.params.{NetworkParams, RegTestParams}
+import com.horizen.params.NetworkParams
 import com.horizen.proof.{Signature25519, VrfProof}
 import com.horizen.proposition.Proposition
-import com.horizen.secret.{PrivateKey25519, VrfSecretKey}
-import com.horizen.transaction.SidechainTransaction
-import com.horizen.utils.{FeePaymentsUtils, ForgingStakeMerklePathInfo, ListSerializer, MerkleTree, TimeToEpochUtils}
-import com.horizen.{SidechainHistory, SidechainMemoryPool, SidechainState, SidechainWallet}
+import com.horizen.secret.PrivateKey25519
+import com.horizen.storage.SidechainHistoryStorage
+import com.horizen.transaction.{SidechainTransaction, Transaction, TransactionSerializer}
+import com.horizen.utils.{DynamicTypedSerializer, FeePaymentsUtils, ForgingStakeMerklePathInfo, ListSerializer, MerklePath, MerkleTree}
+import com.horizen.{SidechainHistory, SidechainMemoryPool, SidechainState, SidechainTypes, SidechainWallet}
 import scorex.core.NodeViewHolder.ReceivableMessages.GetDataFromCurrentView
-import scorex.util.{ModifierId, ScorexLogging}
+import scorex.util.{ModifierId}
 
 import scala.collection.JavaConverters._
-import com.horizen.vrf.VrfOutput
 import scorex.core.NodeViewModifier
+import scorex.core.block.Block.{BlockId, Timestamp}
 
-import scala.collection.mutable.ArrayBuffer
 import scala.util.{Failure, Success, Try}
 
 class ForgeMessageBuilder(mainchainSynchronizer: MainchainSynchronizer,
                           companion: SidechainTransactionsCompanion,
-                          val params: NetworkParams,
-                          allowNoWebsocketConnectionInRegtest: Boolean) extends ScorexLogging {
-  type ForgeMessageType = GetDataFromCurrentView[SidechainHistory, SidechainState, SidechainWallet, SidechainMemoryPool, ForgeResult]
+                          params: NetworkParams,
+                          allowNoWebsocketConnectionInRegtest: Boolean)
+  extends AbstractForgeMessageBuilder[
+    SidechainTypes#SCBT,
+    SidechainBlockHeader,
+    SidechainBlock](
+  mainchainSynchronizer, companion, params, allowNoWebsocketConnectionInRegtest
+) {
+  type HSTOR = SidechainHistoryStorage
+  type VL = SidechainWallet
+  type HIS = SidechainHistory
+  type MS = SidechainState
+  type MP = SidechainMemoryPool
 
-  case class BranchPointInfo(branchPointId: ModifierId, referenceDataToInclude: Seq[MainchainHeaderHash], headersToInclude: Seq[MainchainHeaderHash])
+  type ForgeMessageType = GetDataFromCurrentView[SidechainHistory, SidechainState, SidechainWallet, SidechainMemoryPool, ForgeResult]
```

Comment:
I think, we can put this and `buildForgeMessageForEpochAndSlot` into the base class as well.
we have all the templates known there.

Reply:
Done